Userland/xargs.cpp

```diff
@@ -0,0 +1,276 @@
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <AK/StdLibExtras.h>
+#include <AK/StringBuilder.h>
+#include <LibCore/ArgsParser.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/wait.h>
+
+int run_command(Vector<char*>& child_argv, bool verbose, bool is_stdin, bool devnull_fd);
+
+int main(int argc, char** argv)
+{
+    if (pledge("stdio rpath proc exec", nullptr) < 0) {
+        perror("pledge");
+        return 1;
+    }
+
+    const char* placeholder = nullptr;
+    bool split_with_nulls = false;
+    const char* specified_delimiter = "\n";
+    Vector<const char*> arguments;
+    bool verbose = false;
+    const char* file_to_read = "-";
+    int max_lines_for_one_command = 0;
+    int max_bytes_for_one_command = ARG_MAX;
+
+    Core::ArgsParser args_parser;
+    args_parser.add_option(placeholder, "Placeholder string to be replaced in arguments", "replace", 'I', "placeholder");
+    args_parser.add_option(split_with_nulls, "Split input items with the null character instead of newline", "null", '0');
+    args_parser.add_option(specified_delimiter, "Split the input items with the specified character", "delimiter", 'd', "delim");
+    args_parser.add_option(verbose, "Display each command before executing it", "verbose", 'v');
+    args_parser.add_option(file_to_read, "Read arguments from the specified file instead of stdin", "arg-file", 'a', "file");
+    args_parser.add_option(max_lines_for_one_command, "Use at most max-lines lines to create a command", "line-limit", 'L', "max-lines");
+    args_parser.add_option(max_bytes_for_one_command, "Use at most max-chars characters to create a command", "char-limit", 's', "max-chars");
+    args_parser.add_positional_argument(arguments, "Command and any initial arguments for it", "command", Core::ArgsParser::Required::No);
+    args_parser.parse(argc, argv);
+
+    size_t max_bytes = min(ARG_MAX, max_bytes_for_one_command);
+
+    if (!split_with_nulls && strlen(specified_delimiter) > 1) {
+        fprintf(stderr, "xargs: the delimiter must be a single byte\n");
+        return 1;
+    }
+
+    char entry_separator = split_with_nulls ? '\0' : *specified_delimiter;
+
+    StringView replace_view { placeholder };
+
+    if (arguments.is_empty())
+        arguments.append("echo");
+
+    Vector<Vector<StringView>> argument_parts;
+    argument_parts.ensure_capacity(arguments.size());
+    size_t index = 0;
```

Comment:
I don't see where this is used?

Reply:
curse gcc for not complaining about `++whatever` being unused.
I don't notice this stuff darn it